dwarf/line/line_parser.go

```diff
@@ -33,26 +34,63 @@ type FileEntry struct {
 	Length      uint64
 }
 
-func Parse(data []byte) *DebugLineInfo {
+type DebugLines []*DebugLineInfo
+
+func (d *DebugLines) GetLineInfo(name string) *DebugLineInfo {
+	/*
+		Find in which table file exists.
+		Return that table
+	*/
+	for _, l := range *d {
+		if fe := l.GetFileEntry(name); fe != nil {
+			return l
+		}
+	}
+
+	return nil
+}
+
+func (d *DebugLineInfo) GetFileEntry(name string) *FileEntry {
+	return d.Lookup[name]
+}
+
+func Parse(data []byte) DebugLines {
 	var (
-		dbl = new(DebugLineInfo)
-		buf = bytes.NewBuffer(data)
+		lines = make(DebugLines, 0)
+		buf   = bytes.NewBuffer(data)
 	)
 
-	parseDebugLinePrologue(dbl, buf)
-	parseIncludeDirs(dbl, buf)
-	parseFileEntries(dbl, buf)
-	dbl.Instructions = buf.Bytes()
+	// we have to scan multiple file name tables here
+	for buf.Len() > 0 {
+		dbl := new(DebugLineInfo)
+		dbl.Lookup = make(map[string]*FileEntry)
+
+		parseDebugLinePrologue(dbl, buf)
+		parseIncludeDirs(dbl, buf)
+		parseFileEntries(dbl, buf)
+
+		/*
```

Comment:
Same as comment above regarding prefer line comments. Also, although much appreciated, you can remove the shout out at the end of the calculation description :).